pkg/adsc/delta.go

```diff
@@ -208,7 +208,7 @@ type Client struct {
 
 func (c *Client) trigger(ctx *handlerContext, typeURL string, r *discovery.Resource, event Event) error {
 	res := newProto(typeURL)
-	if r != nil && res != nil {
+	if r != nil && r.Resource != nil && res != nil {
```

Comment:
Maybe instead of passing them a `res T, name string` we should pass them a `Resource[T]`.

Where 

```go
type Resource[T proto.Message] struct {
  Resource T
  Version string
  ResourceName string
  ...
}
```